articles/css-grid/index.md

```diff
@@ -0,0 +1,382 @@
+  <div class="grid-child-8 grid-child">Grid Child 8</div>
+  <div class="grid-child-9 grid-child">Grid Child 9</div>
+  <div class="grid-child-10 grid-child">Grid Child 10</div>
+  <div class="grid-child-11 grid-child">Grid Child 11</div>
+  <div class="grid-child-12 grid-child">Grid Child 12</div>
+  <div class="grid-child-13 grid-child">Grid Child 13</div>
+  <div class="grid-child-14 grid-child">Grid Child 14</div>
+  <div class="grid-child-15 grid-child">Grid Child 15</div>
+</div>
+```
+
+The above code creates a grid of 3 rows of 25%, 50% and 25% size of the parent container, and 5 columns of 100px, 50px, 50px and 100px. The middle column would automatically take up the rest of the space.
+
+![Grid Template Areas](/engineering-education/css-grid/grid-template.png)
+
+We can also use the `fr` unit. One fr or fraction takes up a fraction of the free space available in the grid container.
+
+![Grid Fraction](/engineering-education/css-grid/grid-fr.png)
+
+```html
+<style>
+  .grid-child {
+    background-color: #d0d1cd;
+  }
+
+  .grid-parent-container {
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr 1fr;
+    column-gap: 20px;
+    row-gap: 20px;
+  }
+</style>
+<div class="grid-parent-container">
+  <div class="grid-child-1 grid-child">Grid Child 1</div>
+  <div class="grid-child-2 grid-child">Grid Child 2</div>
+  <div class="grid-child-3 grid-child">Grid Child 3</div>
+  <div class="grid-child-4 grid-child">Grid Child 4</div>
+</div>
+```
+
+In the above code, the `row-gap` defines the gap between grid rows and the `column-gap` defines the gap between grid columns.
+
+The above code creates four columns of one fraction space each. In this case, they each take up 25% space.
+
+### grid-template-areas
+
+This property defines the grid template. It defines which page elements are placed in which grid in the layout.
+
+```css
+.child-1 {
+  grid-area: area-1;
+}
+
+.child-2 {
+  grid-area: area-2;
+}
+
+.child-3 {
+  grid-area: area-3;
+}
+
+.grid-parent-container {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
+  grid-template-rows: 1fr 1fr;
+  grid-template-areas:
+    "area-1 area-2 area-2"
+    "area-1 area-3 area-3";
+}
+```
+
+The above snippet will create a grid that is three columns wide and two rows tall. The top row will be comprised of area-1 and two cells of area-2. The bottom row will be composed of a cell of area-1 and two cells of area-3.
+
+The syntax of `grid-template-areas` provides a visualization of the defined grid structure.
+
+_Note: There is more terminology related to CSS Grid. We have covered only the important ones so that we get a basic understanding of the grid features. For a more detailed overview of all grid features, visit this [MDN link](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout)_
+
+## Let's Code
+
+Our goal is to generate a simple image grid as shown.
+
+![Image Grid](/engineering-education/css-grid/image-grid.png)
```

Comment:
I would emphasise here that the Unsplash images are random so they may not match up with what the user sees.